### PR TITLE
Misc docs edits

### PR DESCRIPTION
**Pull request by olivroy**
The most visible change is that this removes the internal  `dplyr_reconstruct()` from the man page and adds `rename_with.sf()` and `group_split.sf()` which are user facing 

Series of  minor typos and consistency edits throughout.

### CODE REVIEW

File: R/datasets.R [hunk 1]
```diff
@@ -2,8 +2,8 @@
 #'
 #' Sudden Infant Death Syndrome (SIDS) sample data for North Carolina counties,
 #' two time periods (1974-78 and 1979-84). The details of the columns can be
-#' found on the seealso URL, spdep package's vignette. Please note that,
-#' though this is basically the same as \code{nc.sids} dataset in spData
+#' found in a [spdep packages vignette](https://r-spatial.github.io/spdep/articles/sids.html).
```
Comment (olivroy) [inline R/datasets.R:5]:
```suggestion
#' found in a [spdep package vignette](https://r-spatial.github.io/spdep/articles/sids.html). 
```

File: R/shift_longitude.R [hunk 2]
```diff
@@ -6,10 +6,10 @@
 #'   values straddling 180 degrees. In general, using a projected
 #'   coordinate reference system is to be preferred, but this method permits a
 #'   geographical coordinate reference system to be used. This is the sf
-#'   equivalent of \code{\link[sp:recenter-methods]{recenter}} in the sp package and
-#'   ST_ShiftLongitude in PostGIS.
+#'   equivalent of [recenter][sp::recenter] in the sp package and
+#'   `ST_ShiftLongitude` in PostGIS.
 #'
-#' @param x object of class sf or sfc
+#' @param x object of class `sf` or `sfc`.
```
Comment (olivroy) [inline R/shift_longitude.R:12]:
```suggestion
#' @param x object of class `sf` or `sfc`
```

File: R/tidyverse.R [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 # This is currently only used in `bind_rows()` and `bind_cols()`
 # because sf overrides all default implementations
-#' @name tidyverse
+
```
Comment (olivroy) [inline R/tidyverse.R:8]:
```suggestion
```

**Comment by edzer** [comment]
Great, thanks!

**Pull request merged**